arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -795,12 +815,20 @@ public sealed class Option<out A> {
       )
     }
 
+  @Deprecated(
+    NicheAPI + "Prefer when or fold instead",
+    ReplaceWith("fold({ null }) { value -> operation(initial(value), value) }")
```

Comment:
```suggestion
    ReplaceWith("getOrNull()?.let { value -> operation(initial(value), value)  }")
```

Not sure if this is simpler 🤔